dataprofiler/profilers/profile_builder.py

```diff
@@ -502,15 +502,40 @@ def __init__(self, data, samples_per_update=None, min_true_samples=None,
                 utils.warn_on_profile('data_labeler', e)
                 self.options.set({'data_labeler.is_enabled': False})
 
+    def _add_error_checks(self, other):
+        """
+        Profiler type specific checks to ensure two profiles can be added
+        together.
+        """
+        raise NotImplementedError()
+
     def __add__(self, other):
         """
         Merges two profiles together overriding the `+` operator.
 
         :param other: profile being added to this one.
         :type other: BaseProfiler
         :return: merger of the two profiles
+        :rtype: BaseProfiler
         """
-        raise NotImplementedError()
+        if type(other) is not type(self):
+            raise TypeError('`{}` and `{}` are not of the same profiler type.'.
+                            format(type(self).__name__, type(other).__name__))
+
+        # error checks specific to its profiler
+        self._add_error_checks(other)
+
+        merged_profile = self.__class__(
+            data=None, samples_per_update=self._samples_per_update,
+            min_true_samples=self._min_true_samples, options=self.options
+        )
+        merged_profile.encoding = self.encoding \
+            if self.encoding == other.encoding else 'multiple files'
+        merged_profile.file_type = self.file_type \
+            if self.file_type == other.file_type else 'multiple files'
+        merged_profile.total_samples = self.total_samples + other.total_samples
```

Comment:
can we break these into normal if / else looking statements. It's quite difficult to read these multi-line if / else.

Reply:
as in this?
```python
merged_profile.encoding = self.encoding
if self.encoding != other.encoding:
    self.encoding = 'multiple files'

merged_profile.file_type = self.file_type
if self.file_type != other.file_type:
    self.file_type = 'multiple files'
```